onnxruntime/test/optimizer/qdq_transformer_test.cc

```diff
@@ -1210,27 +1210,51 @@ TEST(QDQTransformerTests, DoubleQDQ_Without_Last_Node_Being_Output) {
 // Runs a test that checks if DQ -> Split -> Q (many) is replaced with just Split.
 template <typename InputQType, typename OutputQType>
 static void RunDropSplitQDQTestCase(const std::vector<int64_t>& input_shape, int64_t axis,
-                                    bool use_contrib_qdq = false) {
-  auto check_graph = [use_contrib_qdq](InferenceSessionWrapper& session) {
+                                    bool all_same_quant_params, bool use_contrib_qdq = false) {
+  auto check_graph = [all_same_quant_params, use_contrib_qdq](InferenceSessionWrapper& session) {
     auto op_to_count = CountOpsInGraph(session.GetGraph());
     const QDQOpKeys qdq_keys = GetQDQOpKeys(use_contrib_qdq);
+    int expected_q_ops = all_same_quant_params ? 0 : 3;
+    int expected_dq_ops = all_same_quant_params ? 0 : 1;
     EXPECT_EQ(op_to_count["Split"], 1);
-    EXPECT_EQ(op_to_count[qdq_keys.quantize_linear], 0);
-    EXPECT_EQ(op_to_count[qdq_keys.dequantize_linear], 0);
+    EXPECT_EQ(op_to_count[qdq_keys.quantize_linear], expected_q_ops);
+    EXPECT_EQ(op_to_count[qdq_keys.dequantize_linear], expected_dq_ops);
   };
-  TransformerTester(BuildQDQSplitTestCase<InputQType, OutputQType>(input_shape, axis, use_contrib_qdq),
+  TransformerTester(BuildQDQSplitTestCase<InputQType, OutputQType>(input_shape, axis, !all_same_quant_params,
+                                                                   use_contrib_qdq),
                     check_graph,
                     TransformerLevel::Level1,
                     TransformerLevel::Level2,
-                    {12, 18, 19});
+                    {12, 13, 18, 19});  // Test different ways to specify the split in each opset:
+                                        // 12 - split into equal parts without explicit 'split' attribute
+                                        // 13 - use optional 'split' input to split into 3 parts
```

Comment:
Running this test with an optional 'split' input currently fails in main

Reply:
Is this still the case? The CIs pass so maybe not.

Reply:
This test (opset 13) is not executed on main. I added opset 13 testing in this PR. I'm trying to say that if we ran this new test on main, it would fail (tested locally)